js/util/mapbox.js

```diff
@@ -2,78 +2,87 @@
 
 var config = require('./config');
 var browser = require('./browser');
+var url = require('url');
+var querystring = require('querystring');
 
-function normalizeURL(url, pathPrefix, accessToken) {
+function normalizeURL(parsedUrl, pathPrefix, accessToken) {
```

Comment:
I'm not sure it's helpful to modify `normalizeURL`, it's already able to handle URLs with query strings. We'd only need to make sure that when URLs are manipulated before being passed to `normalizeURL` propagate the query string.